### PR TITLE
Alex Brandt Integrate ChatGPT

### PR DESCRIPTION
**Pull request by brandta-1**
# Description
image

## Related PRS:
This frontend PR is related to the #[809](https://github.com/OneCommunityGlobal/HGNRest/pull/809) backend PR. It was built off [1562](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1562)

### ```src/components/WeeklySummary/WeeklySummary.jsx```
Added ```WriteItForMeModal``` component underneath the ```CurrentPromptModal```

Added ```pasteResponse()``` - a callback function for ```WriteItForMeModal``` that allows the ChatGPT response to be pasted into the editor, with error handling, similar to ```handleEditorChange()```

### ```src/components/WeeklySummary/WriteItForMeModal.jsx```
This button component, when pressed, sends the user's name and id to the server. Then it waits for a response, which contains the text generated by ChatGPT.

## How to test:
1. check into current branch
2. See back-end PR #[809](https://github.com/OneCommunityGlobal/HGNRest/pull/809) for details on setting up the ChatGPT API
3. do `npm install` and `...` to run this PR locally
4. Clear site data/cache
5. log in as any user role
6. create atleast 3 different time entries for this week, the text will be used by ChatGPT:
7. navigate to the Weekly Summary, and press the new Write It For Me button
8. verify that the prompt appeared in the editor box after it loads

## Screenshots or videos of changes:
image
image
image

## Note:
**ChatGPT may return a response that is too short, I have notified Jae.**

### CODE REVIEW

**Review by jennZhang93 — CHANGES_REQUESTED** [review]
I've created an API key and had 3 time logs for testing. During testing with a Core Team account, when clicked on Write It For Me button, the page freezes and the console shows an error of request failed with 404. 

Screenshot 2024-04-09 at 9 40 48 PM

File: src/components/WeeklySummary/WriteItForMeModal.jsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  const fetchSummary = async () => {
+    // eslint-disable-next-line react/destructuring-assignment
+    const { userid } = getUserInfo();
+    const { displayUserProfile } = props;
+    toggle();
+
+    httpService
+      .post(ENDPOINTS.INTERACT_WITH_CHATGPT, {
+        userid,
```
Comment (j-corp-25) [inline src/components/WeeklySummary/WriteItForMeModal.jsx:24]: I looked at the requested payload data. The only data being passed was the name of the user. I did not see userid. Was this behavior happening on your end?

**Review by j-corp-25 — CHANGES_REQUESTED** [review]
Hi @brandta-1 

I encountered a persistent UI issue while implementing the new feature, where the interface stalls indefinitely without any response. Although scrolling is possible, exiting is not.

Here are the steps I took to diagnose the issue:

1. **Network Analysis:**
   - I noticed a **403 error** when attempting to access the endpoint.
   - Additionally, the payload appears incomplete as it only contains the name.
   - This prompted me to review the `WriteItForMeModal` function to understand the cause.

2. **Hardcoding User ID:**
   - Unfortunately, this did not alter the behavior, indicating no new errors.

Attached is a screenshot of the stalled UI for reference:

Testing_PR


I am happy to chat regarding this feature to see why it keeps stalling/freezing the UI.

**Review by metaphor987 — CHANGES_REQUESTED** [review]
Hi @brandta-1, 

I've tested your changes using a volunteer account. After clicking the Write It For Me button, an HTTP error 500 occurred, as shown in the screenshots below:
Screenshot 2024-04-20 at 1 46 42 AM

**Review by KaushikMreddy — CHANGES_REQUESTED** [review]
Hi @brandta-1 

I ran the front end code and backend end code without a hitch, created an API key, added 3 time entries and tried to add a weekly summary. After clicking on write for me, I got this error. 
image

{
    "data": "429 You exceeded your current quota, please check your plan and billing details. For more information on this error, read the docs: https://platform.openai.com/docs/guides/error-codes/api-errors."
}

The CHATGPT API is not free and it has to be paid for. If possible, try to use GEMINI AI API by Google. I believe it is free to use. 


image

Adding onto the previous comment made by @metaphor987, this error shows up with a 500 code, Internal Server Error. So I believe that they are linked.
image


This idea is amazing, would be great if this can be made to work!

**Pull request closed without merging**